app/src/main/java/org/greenstand/android/TreeTracker/managers/LocationUpdateManager.kt

```diff
@@ -120,3 +129,51 @@ fun Location?.accuracyStatus(): Accuracy {
         Accuracy.BAD
     }
 }
+
+class LocationDataCapturer(
+    private val userManager: UserManager,
+    private val locationUpdateManager: LocationUpdateManager,
+    private val treeTrackerDAO: TreeTrackerDAO
+) : KoinComponent {
+
+    init {
+        val gson = Gson()
+        val locationObserver: Observer<Location?> = Observer {
+            it?.apply {
+                MainScope().launch(Dispatchers.IO) {
+                    val locationData =
+                        LocationData(
+                            userManager.planterCheckinId,
+                            latitude,
+                            longitude,
+                            accuracy,
+                            System.currentTimeMillis()
+                        )
+                    val base64String = Base64.encodeToString(
+                        gson.toJson(locationData).toByteArray(),
+                        Base64.NO_WRAP
+                    )
+                    Timber.d("Inserting a new location data $base64String")
+                    treeTrackerDAO.insertLocationData(LocationDataEntity(base64String))
+                }
+            }
+        }
+        locationUpdateManager.locationUpdateLiveData.observeForever(locationObserver)
```

Comment:
Why was the start/stop methods removed? It now starts automatically.

Reply:
Yes, the moment the class is instantiated, it starts listening to the updates from LocationUpdateManager.